<a href="https://colab.research.google.com/github/Azzam-Ar-Pratama/Praktikum-EDA/blob/main/Praktikum_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
# DataFrame - seperti tabel Excel di dalam Python
df = pd.read_csv("data_kantin.csv")
df.head() # lihat 5 baris pertama
df.info() # lihat tipe data & jumlah data tiap kolom
df.describe() # lihat statistik ringkas (rata-rata, min, max, dst)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB


,harga,terjual
count,110.000000,104.000000
mean,6336.363636,22.923077
std,3520.001232,10.831178
min,2000.000000,8.000000
25%,3000.000000,13.000000
50%,5000.000000,20.000000
75%,10000.000000,33.000000
max,12000.000000,45.000000


In [ ]:
import numpy as np
harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


Tugas Analisis 1:
Jika menggunakan NumPy array:
harga = np.array([5000, 7000, 3000])
harga * 0.9
hasilnya: [4500. 6300. 2700.]
Setiap harga langsung dikalikan 0.9.

Sedangkan jika menggunakan list Python biasa:
[5000, 7000, 3000] * 0.9
akan menghasilkan TypeError karena list tidak bisa dikalikan dengan bilangan desimal. Pada list, operator * digunakan untuk mengulang isi list, misalnya list * 2 akan mengulang list sebanyak 2 kali.

In [ ]:
import pandas as pd
data_kantin = {
'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
'harga': [12000, 4000, 10000, 4000, 8000],
'terjual': [23, 40, None, 35, 18]
}
df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


Tugas Analisis 2:
Kolom yang memiliki data kosong (None) adalah:
1. Kolom menu: baris ke-5 kosong.
2. Kolom terjual: baris ke-3 kosong.

Jika data langsung dianalisis tanpa dibersihkan, hasil analisis bisa tidak akurat.

In [ ]:
df = pd.read_csv('data_kantin.csv')
print(df.head()) # 5 baris pertama
print(df.info()) # tipe data & jumlah non-null tiap kolom
print(df.describe()) # statistik ringkas kolom numerik
print(df.shape) # jumlah (baris, kolom)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

Tugas Analisis 3:

Dari hasil df.info(), kolom yang jumlah non-null-nya lebih sedikit dari jumlah baris total (110) adalah:

1. menu: 107 non-null, berarti ada 3 data kosong.
2. terjual: 104 non-null, berarti ada 6 data kosong.

Artinya, pada kolom tersebut terdapat missing value (data yang kosong) sehingga perlu ditangani terlebih dahulu sebelum dilakukan analisis agar hasilnya akurat.

In [ ]:
print(df.isnull().sum()) # jumlah data kosong tiap kolom
df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0
df = df.dropna(subset=['menu']) # hapus baris jika kolom menu kosong

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


Tugas Analisis 4:

Kolom terjual diisi dengan 0 karena jika jumlah penjualan kosong, kita dapat menganggap tidak ada data penjualan atau belum ada yang terjual, sehingga masih bisa digunakan dalam analisis.

Sedangkan baris dengan menu kosong dihapus karena nama menu merupakan informasi penting untuk mengetahui barang yang dijual. Jika nama menunya tidak diketahui, data tersebut sulit digunakan untuk menganalisis penjualan berdasarkan menu.

In [ ]:
print(df.duplicated().sum()) # jumlah baris duplikat
df = df.drop_duplicates()
df['harga'] = df['harga'].astype(int) # memastikan tipe data harga adalah integer
print(df.dtypes)

5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


Tugas Analisis 5:

Sebelum drop_duplicates(), jumlah baris dataset adalah 110 baris.

Setelah menjalankan:

df = df.drop_duplicates()

jumlah baris menjadi 110 dikurangi jumlah baris yang duplikat.
Jadi, jika hasil:

print(df.duplicated().sum())

misalnya 6, maka:

110 − 6 = 104 baris.

In [ ]:
laris = df[df['terjual'] > 20] # filtering
urut = df.sort_values(by='terjual', ascending=False) # sorting

df['total_pendapatan'] = df['harga'] * df['terjual'] # kolom turunan

ringkasan = df.groupby('menu')['total_pendapatan'].sum() # agregasi
print(ringkasan)

Tugas Analisis 6:

Menu yang menghasilkan total_pendapatan tertinggi dapat diketahui dari nilai terbesar pada hasil groupby().

Informasi ini dapat membantu kantin sekolah menentukan menu yang paling menguntungkan, sehingga kantin dapat menyediakan stok lebih banyak, menjaga ketersediaan bahan, dan memprioritaskan menu tersebut untuk meningkatkan pendapatan.

6. Tahap mana (loading, inspection, cleaning, atau manipulation) yang paling menantang bagi saya, dan bagaimana saya mengatasinya?

Tahap yang paling menantang bagi saya adalah cleaning, karena saya harus menangani data kosong dan data duplikat dengan tepat. Saya mengatasinya dengan memeriksa data menggunakan `isnull()` dan `duplicated()`, kemudian menentukan apakah data perlu diisi atau dihapus berdasarkan kondisi dataset.

7. Menurutmu, mengapa keputusan membersihkan data (mis. menghapus vs mengisi data kosong) perlu didasarkan pada alasan yang jelas, bukan asal-asalan?

Karena cara membersihkan data dapat memengaruhi hasil analisis. Jika data dihapus atau diisi secara asal-asalan, hasil analisis bisa menjadi tidak akurat. Oleh karena itu, keputusan cleaning harus disesuaikan dengan kondisi data dan tujuan analisis.

8. Apa hubungan antara dataset bersih hasil proyek ini dengan pekerjaan seorang Data Analyst di dunia nyata?

Dataset yang bersih sangat penting dalam pekerjaan Data Analyst. Seorang Data Analyst perlu memeriksa, membersihkan, dan mengolah data sebelum melakukan analisis agar hasilnya lebih akurat dan dapat dipercaya. Hasil analisis tersebut kemudian dapat digunakan untuk membantu pengambilan keputusan.